src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Repository/ParquetFilterRepository.cs

```diff
@@ -0,0 +1,112 @@
+    IDuckDbConnection duckDbConnection,
+    IParquetPathResolver parquetPathResolver)
+    : IParquetFilterRepository
+{
+    public async Task<IList<ParquetFilterOption>> ListOptions(
+        DataSetVersion dataSetVersion,
+        IEnumerable<int> ids,
+        CancellationToken cancellationToken = default)
+    {
+        return await ListOptionsById<ParquetFilterOption>(
+            dataSetVersion: dataSetVersion,
+            ids: ids.ToList(),
+            columns: ["*"],
+            cancellationToken: cancellationToken
+        );
+    }
+
+    public async Task<IList<ParquetFilterOption>> ListOptions(
+        DataSetVersion dataSetVersion,
+        IEnumerable<string> publicIds,
+        CancellationToken cancellationToken = default)
+    {
+        var publicIdsList = publicIds.ToList();
+
+        if (publicIdsList.Count == 0)
+        {
+            return new List<ParquetFilterOption>();
+        }
+
+        var command = duckDbConnection.SqlBuilder(
+            $"""
+             SELECT *
+             FROM '{parquetPathResolver.FiltersPath(dataSetVersion):raw}'
+             WHERE {FiltersTable.Cols.PublicId:raw} IN ({publicIdsList})
+             """
+        );
+
+        return (await command.QueryAsync<ParquetFilterOption>(cancellationToken: cancellationToken)).AsList();
+    }
+
+    public async Task<IList<IdPublicIdPair>> ListOptionPublicIds(
+        DataSetVersion dataSetVersion,
+        IEnumerable<int> ids,
+        CancellationToken cancellationToken = default)
+    {
+        return await ListOptionsById<IdPublicIdPair>(
+            dataSetVersion: dataSetVersion,
+            ids: ids.ToList(),
+            columns: [FiltersTable.Cols.Id, FiltersTable.Cols.PublicId],
+            cancellationToken: cancellationToken
+        );
+    }
+
+    private async Task<List<T>> ListOptionsById<T>(
+        DataSetVersion dataSetVersion,
+        IList<int> ids,
+        IList<string> columns,
+        CancellationToken cancellationToken = default)
+    {
+        if (ids.Count == 0)
+        {
+            return [];
+        }
+
+        var columnsFragments = new DuckDbSqlBuilder();
+
+        columnsFragments.AppendRange(
+            columns.Select(col => (FormattableString)$"{col:raw}"),
+            joinString: ",\n"
+        );
+
+        var command = duckDbConnection.SqlBuilder(
+            $"""
+             SELECT {columnsFragments}
+             FROM '{parquetPathResolver.FiltersPath(dataSetVersion):raw}'
+             WHERE {FiltersTable.Cols.Id:raw} IN ({ids})
+             """
+        );
+
+        return (await command.QueryAsync<T>(cancellationToken: cancellationToken)).AsList();
+    }
+
+    public async Task<ISet<string>> ListFilterIds(
+        DataSetVersion dataSetVersion,
+        CancellationToken cancellationToken = default)
+    {
+        var command = duckDbConnection.SqlBuilder(
+            $"""
+             SELECT DISTINCT {FiltersTable.Cols.ColumnName:raw}
```

Comment:
@jack-hive penny for your thoughts as I've gone back and forth on this in my head a few times, but what do you think about this column name?

I'm kinda considering renaming it to  `FilterId` (`filter_id` in the table itself). However, this is a touch confusing given the table itself is called `filters.parquet` and we have other id columns too.

Like I said, a bit on the fence about this. It might be fine as-is, but maybe not. Any thoughts?

Reply:
Hmm......

I'm a bit on the fence too.

I think `FilterId` makes more sense in terms of the code - but is less intuitive in terms of the domain context.

Maybe `CsvColumnName` might be more explicit than `ColumnName` and intuitive.

I think I'd either choose `CsvColumnName` or `FilterId`